src/main/java/com/epam/reportportal/service/impl/MigrationServiceImpl.java

```diff
@@ -2,19 +2,25 @@
 
 import com.epam.reportportal.service.ElasticMigrationService;
 import com.epam.reportportal.service.MigrationService;
+import com.epam.reportportal.service.SingleBucketMigrationService;
 import org.springframework.stereotype.Service;
 
 @Service
 public class MigrationServiceImpl implements MigrationService {
 
-	private final ElasticMigrationService elasticMigrationService;
+  private final ElasticMigrationService elasticMigrationService;
 
-	public MigrationServiceImpl(ElasticMigrationService elasticMigrationService) {
-		this.elasticMigrationService = elasticMigrationService;
-	}
+  private final SingleBucketMigrationService singleBucketMigrationService;

```

Comment:
`@Autowired
private final List<MigrationService>
`
should return list of implementations of MigrationService
then call migrate for all services
migrations.forEach(service -> service.migrate())

Should be more scalable